xbmc/music/Album.cpp

```diff
@@ -53,30 +53,38 @@ CAlbum::CAlbum(const CFileItem& item)
   strMusicBrainzAlbumID = tag.GetMusicBrainzAlbumID();
   genre = tag.GetGenre();
   artist = tag.GetAlbumArtist();
+  std::vector<std::string> musicBrainAlbumArtistHints = tag.GetMusicBrainzAlbumArtistHints();
+  strArtistDesc = tag.GetAlbumArtistDesc();
+
   if (!tag.GetMusicBrainzAlbumArtistID().empty())
   { // have musicbrainz artist info, so use it
     for (size_t i = 0; i < tag.GetMusicBrainzAlbumArtistID().size(); i++)
     {
       std::string artistId = tag.GetMusicBrainzAlbumArtistID()[i];
       std::string artistName;
-      /*
-       We try and get the corresponding artist name from the album artist tag.
-       We match on the same index, and if that fails just use the first name we have.
-       If no albumartist exists, try matching on artist if the MBArtistID matches.
-       */
-      if (!artist.empty())
-        artistName = (i < artist.size()) ? artist[i] : artist[0];
+
+      if ( i < musicBrainAlbumArtistHints.size() )
```

Comment:
Is the comment to this part obsolete or why have you removed it? It would probably be nicer to just update the comment